nodes/test/test_pinecone_record_updates.py

```
@@ -0,0 +1,356 @@
+def _make_store(records: dict[str, dict]) -> tuple[object, _FakeIndex]:
+    """Create a store instance with fake Pinecone client/index."""
+    store_class = _load_store_class()
+    fake_index = _FakeIndex(records)
+    store = store_class.__new__(store_class)
+    store.collection = 'test-index'
+    store.client = _FakeClient(fake_index)
+    store.doesCollectionExist = lambda *_args, **_kwargs: True
+    return store, fake_index
+
+
+def _make_obj_records(object_id: str, count: int, is_deleted: bool) -> dict[str, dict]:
```

Ruff (0.15.6)

[warning] Boolean-typed positional argument in function definition

(FBT001)